micro-rdk/src/common/app_client.rs

```diff
@@ -255,19 +255,18 @@ impl AppClient {
     // `last_reconfigured` values for resource statuses.
     pub async fn get_config(
         &self,
-        ip: Ipv4Addr,
+        ip: Option<Ipv4Addr>,
     ) -> Result<(Box<ConfigResponse>, Option<DateTime<FixedOffset>>), AppClientError> {
-        let agent = AgentInfo {
+        let agent = ip.map(|ip|AgentInfo {
             os: "esp32".to_string(),
             host: "esp32".to_string(),
             ips: vec![ip.to_string()],
             version: env!("CARGO_PKG_VERSION").to_string(),
             git_revision: "".to_string(),
             platform: Some("esp32".to_string()),
-        };
-
+        });
         let req = ConfigRequest {
-            agent_info: Some(agent),
+            agent_info: agent,
```

Comment:
What are the practical effects of the `agent_info` being `None` here when invoking the `/viam.app.v1.RobotService/Config` endpoint, if any?

Reply:
<img width="343" alt="image" src="https://github.com/viamrobotics/micro-rdk/assets/83091926/295f2387-b571-44dd-b286-10bf6bf75125">
It's used to show app info, but if we don't supply it, it should still work fine